From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: xKumorio <78680226+xKumorio@users.noreply.github.com>
Date: Sat, 2 Aug 2025 03:02:28 +0500
Subject: [PATCH] lobotomize


diff --git a/net/minecraft/world/entity/boss/wither/WitherBoss.java b/net/minecraft/world/entity/boss/wither/WitherBoss.java
index 787b74c5aa02afc4ba95fa1cdaf6cc21b6554b56..e452bca994d93d8a5fc73d692bea10234e3f92b0 100644
--- a/net/minecraft/world/entity/boss/wither/WitherBoss.java
+++ b/net/minecraft/world/entity/boss/wither/WitherBoss.java
@@ -103,7 +103,7 @@ public class WitherBoss extends Monster implements RangedAttackMob {
         this.goalSelector.addGoal(6, new LookAtPlayerGoal(this, Player.class, 8.0F));
         this.goalSelector.addGoal(7, new RandomLookAroundGoal(this));
         this.targetSelector.addGoal(1, new HurtByTargetGoal(this));
-        this.targetSelector.addGoal(2, new NearestAttackableTargetGoal<>(this, LivingEntity.class, 0, false, false, LIVING_ENTITY_SELECTOR));
+        this.targetSelector.addGoal(2, new NearestAttackableTargetGoal<>(this, net.minecraft.world.entity.Mob.class, 0, false, false, WitherBoss.LIVING_ENTITY_SELECTOR));
     }
 
     @Override
